elastic.go

```diff
@@ -18,16 +19,18 @@ type Handler struct {
 	typ    string
 	// Refresh sets the refresh flag to true on all write operation to ensure
 	// writes are reflected into search results immediately after the operation.
-	// Setting this parameter to true has performance impacts.
-	Refresh bool
+	// Setting this parameter to "true" has performance impacts.
+	Refresh string
```

Comment:
Making this public parameter a string seams off to me. Now it might hold any value, including "whatever", which just makes the API more error prone in usage. It would be better to translate it to a string when needed.

If it's because we need to support more than one value, make a private (or public) type with const values:
```go
type refreshVal string

const (
   RefreshTrue = refreshVal("true")
   RefreshFalse = refreshVal("false")
   RefreshSometimes = refreshVal("sometimes")
)

type Handler {
    ...
    Refresh refreshVal // This might give you linter warnings, due to the private type
                       // but would ensure "enum" values are used outside the package.
                       // Alternatively a public time is almost as safe as people would need
                       // an explicit type conversion to use it wrong.
    ...
}
```

If it's a performance issue, find some way to deal with it that doesn't show in the public API. E.g. use a lookup table:

```go
var fereshMap = map[bool]string{
true: "true",
false: "false"
}
```